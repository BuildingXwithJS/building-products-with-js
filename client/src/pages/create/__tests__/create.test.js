// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import CreatePage, {Create} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
  registrationDate: new Date(2016, 1, 1, 1, 1, 1, 1),
};

test('# Create page wrapper', () => {
  const store = mockStore({auth: {user}});
  const wrapper = shallow(<CreatePage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Create page', () => {
  const newText = 'newText';
  const newExpirationDate = new Date(2016, 1, 1, 1, 1, 1, 1);
  const doCreateQuestion = ({text, expirationDate}) => {
    expect(text).toBe(newText);
    expect(expirationDate).toBe(newExpirationDate.toISOString());
  };

  const component = (
    <Create
      user={user}
      doCreateQuestion={doCreateQuestion}
    />
  );

  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // test user update
  const app = mount(component);
  // set new question text
  app.find('#questionText').getDOMNode().value = newText;
  // set new question expiration date
  app.find('#expirationDate').getDOMNode().value = newExpirationDate.toISOString();
  // click answer button
  app.find('button').simulate('click');
});
