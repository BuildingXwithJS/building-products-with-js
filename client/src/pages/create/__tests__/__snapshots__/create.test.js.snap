// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Create page 1`] = `
<div>
  <Component
    current="/create"
    user={
      Object {
        "id": "0",
        "login": "test",
        "registrationDate": 2016-02-01T00:01:01.001Z,
      }
    }
  />
  <div>
    <form>
      <div
        className="form-group"
      >
        <label
          htmlFor="questionText"
        >
          Question text
        </label>
        <input
          className="form-control"
          id="questionText"
          placeholder="Question text"
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="expirationDate"
        >
          Expiration date
        </label>
        <input
          className="form-control"
          id="expirationDate"
          placeholder="Expiration date"
          type="date"
        />
      </div>
      <button
        className="btn btn-default"
        onClick={[Function]}
        type="submit"
      >
        Create new question
      </button>
    </form>
  </div>
</div>
`;

exports[`# Create page wrapper 1`] = `
<Create
  doCreateQuestion={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  user={
    Object {
      "id": "0",
      "login": "test",
      "registrationDate": 2016-02-01T00:01:01.001Z,
    }
  }
/>
`;
